### PR TITLE
Intern `CanonicalVarValues`

### PR DESCRIPTION
**Pull request by compiler-errors**
So that they are copy :sparkles:

### CODE REVIEW

**Comment by compiler-errors** [comment]
Wonderful

File: compiler/rustc_trait_selection/src/solve/mod.rs [hunk 1]
```diff
@@ -454,26 +452,24 @@ pub(super) fn response_no_constraints<'tcx>(
...
+                CanonicalVarKind::Const(_, ty) | CanonicalVarKind::PlaceholderConst(_, ty) => tcx
+                    .mk_const(ty::ConstKind::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i)), ty)
+                    .into(),
             }
-            CanonicalVarKind::Const(_, ty) | CanonicalVarKind::PlaceholderConst(_, ty) => tcx
-                .mk_const(ty::ConstKind::Bound(ty::INNERMOST, ty::BoundVar::from_usize(i)), ty)
-                .into(),
-        })
-        .collect();
+        }));
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/solve/mod.rs:472]: :thinking: this should just be `CanonicalVarValues::make_identity`? so we should replace it

File: compiler/rustc_trait_selection/src/solve/search_graph/cache.rs [hunk 2]
```diff
@@ -95,6 +95,7 @@ impl<'tcx> ProvisionalCache<'tcx> {
     }
 
     pub(super) fn provisional_result(&self, entry_index: EntryIndex) -> QueryResult<'tcx> {
+        // FIXME: Responses should probably be `Copy` as well
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/solve/search_graph/cache.rs:98]: ah yeah, probably want to also intern `ExternalConstraints` or something :thinking:

**Comment by lcnr** [comment]
r=me after nit :heart:

File: compiler/rustc_trait_selection/src/solve/mod.rs (hunk 1 above)
Comment (compiler-errors) [reply compiler/rustc_trait_selection/src/solve/mod.rs:472]: But we don't have a CanonicalVarValues here, we have a CanonicalVarInfos, so we can't call make_identity?

File: compiler/rustc_trait_selection/src/solve/search_graph/cache.rs (hunk 2 above)
Comment (compiler-errors) [reply compiler/rustc_trait_selection/src/solve/search_graph/cache.rs:98]: Yeah i'm gonna do that in a follow up, then we can remove the other clones :D

File: compiler/rustc_trait_selection/src/solve/mod.rs (hunk 1 above)
Comment (compiler-errors) [reply compiler/rustc_trait_selection/src/solve/mod.rs]: Actually, there are no call-sites for `make_identity` anymore :thinking: 

I'll just make it take `CanonicalVarInfos`

**Comment by compiler-errors** [comment]
@bors r=lcnr

nit addressed

**Pull request merged**